### PR TITLE
fix(html): improve DOM structure

### LINKED ISSUE #1700: [Code Quality] Some instances of the HTML main tag are not hierarchically correct
### Steps to reproduce the problem

Take a look at https://github.com/Qiskit/qiskit.org/pull/1699/files#r615520384

### What is the current behavior?

Some `main` tags appear inside the `header` tag. This is incorrect attending to the [HTML specification](https://html.spec.whatwg.org/#hierarchically-correct-main-element).

### What is the expected behavior?

All `main` occurrences should be hierarchically correct.

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes

Closes #1700 
Co-authored by: @mariacloehb 

## Implementation details

- remove all instances of having 2 `<main>` elements in various pages or layouts
- restructure the DOM so to use this clear pattern:

```html
<html>
  <body>
    <div id="__nuxt">
      <header></header>
      <main></main>
      <footer></footer>
    </div>
  </body>
</html>
```

## Preview link

https://qiskit-org-pr-3443.dcq4xc5i083.us-south.codeengine.appdomain.cloud/ecosystem

## How to read this PR

- in `LayoutLeadspaceWithCard`, please review the renaming of <main> and `&__main` to `<div>` and `&__hero`
- please provide feedback on new structure, using `<header>` `<main>` and `<footer>`
  - now there's only one `<main>` throughout the entire project

## Screenshots

1) Removing second `<main>` element in `LayoutLeadspaceWithCard` component
leadspace-with-card-layout

<br/>

2) Removing second `<main>` element in Learn pages
learn-pages

<br/>

3) Restructuring HTML
dom-restructure

### CODE REVIEW

File: components/LearnCourse/LearnCourse.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main class="learn-course">
+  <div class="learn-course">
```
Comment (y4izus) [inline components/LearnCourse/LearnCourse.vue:2]: Could you double check this change? I think this is more an [article](https://html.spec.whatwg.org/#the-article-element) than a [div](https://html.spec.whatwg.org/#the-div-element)

File: pages/advocates.vue [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main>
+  <div>
```
Comment (y4izus) [inline pages/advocates.vue:2]: Same as before, could you double check if this fits better as `article` than a `div`?

File: pages/ecosystem.vue [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main>
+  <div>
```
Comment (y4izus) [inline pages/ecosystem.vue:2]: Same, `article` vs `div`?

File: pages/events/fall-fest.vue [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main class="fall-fest-page">
+  <div class="fall-fest-page">
```
Comment (y4izus) [inline pages/events/fall-fest.vue:2]: Same, `article`?

File: pages/events/seminar-series.vue [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main>
+  <div>
```
Comment (y4izus) [inline pages/events/seminar-series.vue:2]: Same, `article`?

File: pages/events/summer-school-2022.vue [hunk 6]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main class="summer-school-2022-page">
+  <div class="summer-school-2022-page">
```
Comment (y4izus) [inline pages/events/summer-school-2022.vue:2]: Same, `article`?

File: pages/events/summer-school-2023.vue [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <main class="summer-school-2023-page">
+  <div class="summer-school-2023-page">
```
Comment (y4izus) [inline pages/events/summer-school-2023.vue:2]: Same, `article`?

File: pages/index.vue [hunk 8]
```diff
@@ -1,10 +1,10 @@
 <template>
-  <main>
+  <div>
```
Comment (y4izus) [inline pages/index.vue:2]: Same, `article`?

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you!! Could you review my comments? I think we can improve the use of the HTML

File: pages/learn/index.vue [hunk 9]
```diff
@@ -1,12 +1,12 @@
 <template>
-  <main class="learn-page">
+  <div class="learn-page">
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply pages/learn/index.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: pages/index.vue (hunk 8 above)
Comment (techtolentino) [reply pages/index.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: pages/events/summer-school-2023.vue (hunk 7 above)
Comment (techtolentino) [reply pages/events/summer-school-2023.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: pages/events/summer-school-2022.vue (hunk 6 above)
Comment (techtolentino) [reply pages/events/summer-school-2022.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: pages/events/seminar-series.vue (hunk 5 above)
Comment (techtolentino) [reply pages/events/seminar-series.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: pages/events/fall-fest.vue (hunk 4 above)
Comment (techtolentino) [reply pages/events/fall-fest.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: pages/ecosystem.vue (hunk 3 above)
Comment (techtolentino) [reply pages/ecosystem.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: pages/advocates.vue (hunk 2 above)
Comment (techtolentino) [reply pages/advocates.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

File: components/LearnCourse/LearnCourse.vue (hunk 1 above)
Comment (techtolentino) [reply components/LearnCourse/LearnCourse.vue]: Updated in https://github.com/Qiskit/qiskit.org/pull/3441/commits/8813034834a0b3ef57c06bdf5a74df881100161f

**Review by y4izus — APPROVED** [review]

**Pull request merged**